chainio/utils/bindings.go

```diff
@@ -16,20 +18,26 @@ import (
 	strategymanager "github.com/Layr-Labs/eigensdk-go/contracts/bindings/StrategyManager"
 )
 
-type eigenlayerContractBindings struct {
-	Slasher             *slasher.ContractSlasher
-	DelegationManager   *delegationmanager.ContractDelegationManager
-	StrategyManager     *strategymanager.ContractStrategyManager
-	BlsPubKeyCompendium *blspubkeycompendium.ContractBLSPublicKeyCompendium
-	logger              logging.Logger
+// Unclear to me why geth bindings don't store and expose the contract address...
+// so we also store them here in case the different constructors that use this struct need them
+type EigenlayerContractBindings struct {
+	SlasherAddr             gethcommon.Address
+	StrategyManagerAddr     gethcommon.Address
+	DelegationManagerAddr   gethcommon.Address
+	BlspubkeyCompendiumAddr gethcommon.Address
+	Slasher                 *slasher.ContractSlasher
+	DelegationManager       *delegationmanager.ContractDelegationManager
+	StrategyManager         *strategymanager.ContractStrategyManager
+	// TODO: should this really be here? or moved to avs bindings
+	BlsPubkeyCompendium *blspubkeycompendium.ContractBLSPublicKeyCompendium
```

Comment:
I am very confused with this - seems like we deploy it with AVS today (with EigenDA deployment). how is it going to work when we have multiple AVS?

Reply:
We originally were thinking to have a single one shared between all teams. But I think the current thinking is that every team will deploy their own. I'm in favor of this change because it gives more freedom to operators to use different keys for different avss, albeit costs a bit more gas.

Reply:
so if every AVS has different BLS pubkey compendium then we need to rethinking combining BLS pubkey registration via EigenLayer CLI since the current assumption was that there's only one.